astropy/io/misc/asdf/tags/transform/basic.py

```diff
@@ -90,6 +96,16 @@ def _to_tree_base_transform_members(cls, model, node, ctx):
             if bounds_nondefaults:
                 node['bounds'] = bounds_nondefaults
 
+        if not isinstance(model, CompoundModel):
+            if model.input_units is not None:
+                if model.input_units_equivalencies:
+                    input_unit_equivalencies = {}
+                    for in_unit in model.input_units:
+                        eq = model.input_units_equivalencies[in_unit]
```

Comment:
It's possible for equivalencies to be specified for only a subset of inputs, right?  In that case, this looks like it'll raise a KeyError.

Reply:
This has been superseded by https://github.com/astropy/astropy/pull/10198#discussion_r768877413.